File: api/src/main/java/net/impactdev/gts/api/messaging/message/type/utility/MaintenanceUpdateMessage.java
package net.impactdev.gts.api.messaging.message.type.utility;

import net.impactdev.gts.api.maintenance.MaintenanceMode;
import net.impactdev.gts.api.messaging.message.OutgoingMessage;
import net.impactdev.gts.api.messaging.message.type.UpdateMessage;

public interface MaintenanceUpdateMessage extends UpdateMessage, OutgoingMessage {

    MaintenanceMode getMode();

    boolean getState();

}

File: api/src/main/kotlin/net/impactdev/gts/api/messaging/message/type/utility/MaintenanceUpdateMessage.kt
package net.impactdev.gts.api.messaging.message.type.utility

import net.impactdev.gts.api.maintenance.MaintenanceMode
import net.impactdev.gts.api.messaging.message.OutgoingMessage
import net.impactdev.gts.api.messaging.message.type.UpdateMessage

interface MaintenanceUpdateMessage : UpdateMessage, OutgoingMessage {
    val mode: MaintenanceMode?
    val state: Boolean
}